### DISCUSSION TITLE
Set aria-current attributes on next/link

### DISCUSSION THREAD

**Discussion by alexandernanberg** [vercel/next.js#6014] [Ideas]
# Feature request

## Is your feature request related to a problem? Please describe.
Right now you have to do this yourself and I'd bet many forget this. It would be really nice to have this included in `next/link` by default.

## Describe the solution you'd like
Set a `aria-current` attribute on the active link(s). I could work on this, but I might need some guidance on how this should be implemented since we probably have to wrap the link component in a `withRoute` hoc.

## Describe alternatives you've considered
N/A

## Additional context
https://www.w3.org/TR/wai-aria-1.1/#aria-current

**Comment by hnavarro32** [comment]
Forgive I don't quite know your use case but I think what you are looking for is something like this:

```
const NavLink = styled.a`...`;
...
<Link href="/about/[entity]" as={"/about/someEntityName"} passHref> //passHref since I am assuming you're using styled components
    <NavLink aria-label="Go read more about this entity here" />
</Link>
```

This will result in Nextjs creating an `<a>` tag with the appropriate aria tag you want to add without any cruft in the next link component.

I found this page in the documentation helpful https://nextjs.org/docs/api-reference/next/link#if-the-child-is-a-custom-component-that-wraps-an-a-tag

**Comment by ShaneMcX** [comment]
I'm not sure, but I think the OP was looking for a way to specifically use aria-current, as I am. Specifically, it would be great if aria-current=page would be set if the link points to the current page.

https://amanexplains.com/using-aria-current-for-nav-links/ describes a way to do this, though it only works on client components as far as I can tell. (there's discussion of a server approach there, but it's not working for me with NextJS 13.
